Use an apt-key alternative in install instructions

ptgott opened the pull request:
Current installation instructions for Debian/Ubuntu advise using
apt-key, which is deprecated. This change uses the recommended
method of appending a "signed-by" argument to the APT sources
list along with the rest of a repo's information.

Also creates a partial with Linux installation info to avoid
having to repeat changes in our different Linux installation
instructions.

Fixes #9611

ptgott committed:
Use an apt-key alternative in install instructions

Current installation instructions for Debian/Ubuntu advise using
apt-key, which is deprecated. This change uses the recommended
method of appending a "signed-by" argument to the APT sources
list along with the rest of a repo's information.

Also creates a partial with Linux installation info to avoid
having to repeat changes in our different Linux installation
instructions.

Fixes #9611

zmb3 commented on docs/pages/includes/install-linux.mdx:
```diff
@@ -0,0 +1,65 @@
+        # Verify that the checksums match
+        tar -xzf teleport-v(=teleport.version=)-linux-amd64-bin.tar.gz
+        cd teleport
+        ./install
```
The install script requires root.
```suggestion
        sudo ./install
```

ptgott committed:
Respond to PR feedback

- Add "sudo" before the installation script
- Add command prompts before commands (not part of PR feedback but
  something I noticed was missing)

modem7 commented:
@ptgott - I attempted to cat the file, but got access denied. 

Seems that it only worked if I was root. 

Non-root w/ sudo access
![image](https://user-images.githubusercontent.com/4349962/153288807-20a4cb07-84f7-4e35-b396-a52489baab9e.png)

sudo -i:
![image](https://user-images.githubusercontent.com/4349962/153288937-c0a356f4-05c2-44db-bea5-f0c2101fe1b6.png)


This method seems to work without being su first however:

```
sudo curl https://deb.releases.teleport.dev/teleport-pubkey.asc \
          -o /usr/share/keyrings/teleport-archive-keyring.asc && \
sudo touch /etc/apt/sources.list.d/teleport.list && \
echo 'deb [signed-by=/usr/share/keyrings/teleport-archive-keyring.asc] https://deb.releases.teleport.dev/ stable main' | sudo tee -a /etc/apt/sources.list.d/teleport.list && \
sudo apt-get update && \
sudo apt-get install teleport
```
